src/data_visualization/src/main.rs

```diff
@@ -1,26 +1,76 @@
-use std::{error::Error, process};
-
-use polars::frame::DataFrame;
-use polars::prelude::*;
+use std::error::Error;
 
 mod histogram;
 mod pair_plot;
 mod parse_cols;
 mod scatter_plot;
 
+use clap::{arg, command, value_parser};
+
 fn main() -> Result<(), Box<dyn Error>> {
-	let filename = "./ressources/dataset_train.csv";
-	let Ok(data) = parse_cols::load_as_cols(filename) else {
+	let args = command!() // requires `cargo` feature
+		.arg(
+			arg!(--csv <csvname>)
+				.help("Houses csv path")
+				.value_parser(value_parser!(String))
+				.default_value("./ressources/dataset_train.csv")
+				.required(false),
+		)
+		.arg(
+			arg!(--scatter <scatter_path>)
+				.help("Scatter plot name")
+				.value_parser(value_parser!(String))
+				.default_value("Scatter_plot.png")
+				.required(false),
+		)
+		.arg(
+			arg!(--hist <hist_path>)
+				.help("Histogram plot name")
+				.value_parser(value_parser!(String))
+				.default_value("Histogram.png")
+				.required(false),
+		)
+		.arg(
+			arg!(--pair <pair_path>)
+				.help("Pair Plot plot name")
+				.value_parser(value_parser!(String))
+				.default_value("Pair_Plot.png")
+				.required(false),
+		)
+		.get_matches();
+
+	let filename = args
+		.get_one::<String>("csv")
+		.expect("default ensures there is always a value");
+	let Ok(mut data) = parse_cols::load_as_cols(filename) else {
 		return Err("Invalid file".into());
 	};
 
-	let data = parse_cols::transform_data(data)?;
+	// TODO: polars_core::frame::DataFrame and
+	// polare::prelude::DataFrame clash, but both seem to be the same
+	// Alex sos
+	// the rest works though
+	// let mut data = load::load(filename)?;
```

Comment:
first: the version of `polars` is different between the 2 library
- we should be using `0.43` (latest)

second: prefer `polars::prelude::DataFrame`
- if a `lib` has `prelude`, prefer using import from it